lib/langs/python.rb

```diff
@@ -67,10 +94,56 @@ def parse_pipfile(content)
     def latest_version(name)
       content = HTTP.follow(max_hops: 1).get(::File.join(API_URL, name, 'json'))
       begin
-        Gem::Version.new(JSON.parse(content)['info']['version'])
-      rescue
+        v = JSON.parse(content)['info']['version']
+        Gem::Version.new(canonical_version(v))
+      rescue StandardError => e
+        log_error("Encountered error when fetching latest version of #{name}: #{e.class} #{e.message}")
         nil
       end
     end
+
+    private
+
+    def get_type(type)
+      type == 'dev-packages' ? :development : :runtime
+    end
+
+    def canonical_version(v)
+      case v
+      when /^\./
+        "0#{v}"
+      else
+        v
+      end
+    end
+
+    # Translate Pipfile syntax to a String or Array of Strings that Gem::Dependency.new understands
+    # See https://www.python.org/dev/peps/pep-0440/#version-matching
+    def translate_requirement(req)
+      req.sub!('~=', '~>')
+      req.sub!('==', '=')
+      case req
+      when '*'
+        '>= 0'
+      when /\.\*/
+        if req =~ /!=/
+          # Turn != 1.1.* into >= 1.2 OR < 1.1
+          req.sub!('.*', '.0')
+          req.sub!('!=', '')
+          v = Gem::Version.new(req) # Create the bump version using Gem::Version so pre-release handing will work
+          lower_bound = ">= #{v.bump.version}"
```

Comment:
SemVer refers to `bump` as `patch` version. Perhaps we want to consistently use patch as well throughout the code base.

Reply:
I've changed `outdated_bump` to `outdated_patch`. (In this particular context, `bump` is the designation used by rubygems.)